src/binary/ibc-setup/commands/channel.ts

```diff
@@ -0,0 +1,154 @@
+  readonly home: string;
+  readonly mnemonic: string;
+  readonly src: string;
+  readonly dest: string;
+  readonly srcConnection: string;
+  readonly destConnection: string;
+  readonly srcPort: string;
+  readonly destPort: string;
+  readonly version: string;
+  readonly ordered: boolean;
+};
+
+export async function channel(flags: Flags, logger: Logger) {
+  const home = resolveHomeOption({ homeFlag: flags.home });
+  const app = loadAndValidateApp(home);
+
+  const keyFile = resolveKeyFileOption({ keyFileFlag: flags.keyFile, app });
+  const mnemonic = await resolveMnemonicOption({
+    interactiveFlag: flags.interactive,
+    mnemonicFlag: flags.mnemonic,
+    keyFile: keyFile,
+    app,
+  });
+  const src = resolveOption('src', { required: true })(
+    flags.src,
+    app?.src,
+    process.env.RELAYER_SRC
+  );
+  const dest = resolveOption('dest', { required: true })(
+    flags.dest,
+    app?.dest,
+    process.env.RELAYER_DEST
+  );
+  const srcConnection = resolveOption('srcConnection', { required: true })(
+    flags.srcConnection,
+    app?.srcConnection,
+    process.env.RELAYER_SRC_CONNECTION
+  );
+  const destConnection = resolveOption('destConnection', { required: true })(
+    flags.destConnection,
+    app?.destConnection,
+    process.env.RELAYER_DEST_CONNECTION
+  );
+  const srcPort = resolveOption('srcPort', { required: true })(
+    flags.srcPort,
+    process.env.RELAYER_SRC_PORT
+  );
+  const destPort = resolveOption('destPort', { required: true })(
+    flags.destPort,
+    process.env.RELAYER_DEST_PORT
+  );
+  const version =
+    resolveOption('version')(flags.version, process.env.RELAYER_VERSION) ??
+    'ics20-1';
+
+  const options: Options = {
+    home,
+    mnemonic,
+    src,
+    dest,
+    srcConnection,
+    destConnection,
+    srcPort,
+    destPort,
+    version,
+    ordered: flags.ordered,
+  };
+
+  await run(options, logger);
+}
+
+export async function run(options: Options, logger: Logger) {
+  const registryFilePath = path.join(options.home, registryFile);
+  const registry = loadAndValidateRegistry(registryFilePath);
+  const srcChain = registry.chains[options.src];
+  if (!srcChain) {
+    throw new Error(`src channel "${options.src}" not found in registry`);
+  }
+  const destChain = registry.chains[options.dest];
+  if (!destChain) {
+    throw new Error(`dest channel "${options.dest}" not found in registry`);
+  }
+
+  const nodeA = await signingClient(
+    srcChain,
+    options.mnemonic,
+    logger.child({ label: srcChain.chain_id })
+  );
+  const nodeB = await signingClient(
+    destChain,
+    options.mnemonic,
+    logger.child({ label: destChain.chain_id })
+  );
+
+  const link = await Link.createWithExistingConnections(
+    nodeA,
+    nodeB,
+    options.srcConnection,
+    options.destConnection,
+    logger
+  );
+
+  const ordering = options.ordered
+    ? Order.ORDER_ORDERED
+    : Order.ORDER_UNORDERED;
+
+  const channels = await link.createChannel(
```

Comment:
How about renaming this method `createChannels`?

Reply:
It is only one channel, with 2 endpoints.
Maybe the variable needs to be renamed?

Reply:
cd2d00b

I renamed the variable to `channel`, however, in the stdout we still print:
```
Created channels for connections  (...)
```

and that's confusing.

Should we change it?
```
Created channel with endpoints for connections (...)
```

or

```
Created endpoints for connections (...)
```

Reply:
How about this (also making it easy to parse as mentioned below):

```
Created channel:
 <src-chain-id>: <src-port-id>/<src-channel-id> (<src-connection-id>) 
 <dest-chain-id>: <dest-port-id>/<dest-channel-id> (<dest-connection-id>)
```

In other tests you can match a line by chain-id, then easily parse out needed info. Note that channel-id's are theoretically not global identifiers (although in practice I think so), so you should always reference them by port/channel to make them easy to query/find later.

Reply:
386339c

Great 👍 
<img width="413" alt="image" src="https://user-images.githubusercontent.com/79154001/113775140-8f15f900-9728-11eb-9e01-c350a4327dfb.png">